### PR TITLE
Remove api keys and dnt headers from Pelias, replace with PeliasService

### PR DESCRIPTION
**Pull request by sarahlensing**
Api key can now be set via the `RestAdapter` used to create the `PeliasService` using a `RequestInterceptor` like this:

```
RestAdapter restAdapter = new RestAdapter.Builder().setEndpoint(
        Pelias.DEFAULT_SEARCH_ENDPOINT).setRequestInterceptor(new RequestInterceptor() {
      @Override public void intercept(RequestFacade request) {
        request.addQueryParam("api_key", "EXAMPLE_API_KEY");
      }
    }).build();
    pelias.setService(restAdapter.create(PeliasService.class));
```

### CODE REVIEW

**Comment by ecgreb** [comment]
I think instantiation of the `RestAdapter` should remain the responsibility for the `Pelias` class. This seems like an implementation detail we should keep hidden from the client.

So the the question becomes how to append an arbitrary query param (i.e. API key) or header (i.e. DNT) to the request if the service object is not exposed?

One possibility is to make a version of the `Pelias` constructor that accepts a `RequestInterceptor` (or a custom interface) rather than a `PeliasService` but there might be a more elegant solution.

**Comment by sarahlensing** [comment]
I've updated this to keep the instantiation of `RestAdapter` within `Pelias` and added an interface for setting query params and headers on requests.

**Comment by ecgreb** [comment]
Changes look good! Only things I noticed were two unused imports in `PeliasTest`. Also the field `String apiKey = "test_key"` is no longer needed and can be removed.

Created an issue to add `checkstyle` to the project ;)

https://github.com/pelias/pelias-android-sdk/issues/45

File: src/test/java/com/mapzen/pelias/PeliasTest.java [hunk 1]
```diff
@@ -13,7 +13,12 @@
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import retrofit.Callback;
+import retrofit.RequestInterceptor;
```
Comment (ecgreb) [inline src/test/java/com/mapzen/pelias/PeliasTest.java]: Unused imports

**Comment by sarahlensing** [comment]
Nice, that will be good to have.

File: src/test/java/com/mapzen/pelias/PeliasTest.java (hunk 1 above)
Comment (sarahlensing) [inline src/test/java/com/mapzen/pelias/PeliasTest.java]: Rm'd

**Pull request merged**